src/trunk.c

```diff
@@ -1230,6 +1230,13 @@ trunk_generation(trunk_handle *spl, page_handle *node)
    return hdr->generation;
 }
 
+static inline uint16
+trunk_pivot_start_bundle(trunk_handle *spl, page_handle *node, uint16 pivot_no)
```

Comment:
About this collection of changes around trunk-bundle mgmt, some comments (food for thought, for follow-up):

1. What tests ran into bugs that you were fixing in this small set of changes? Are those tests already included in CI- or nightly-tests? If not, what is our plan / approach to augment test run to 'discover' those bugs, and to ensure that these changes do fix them. 

(Point being, can we plan better to integrate test-changes along with code-changes, especially for such bug-fix type code-changes,  so that we are continually improving code coverage thru tests.)

2. I noted the addition of small interface like this `trunk_pivot_start_bundle(). And found few more existing minimalist ones like`trunk_start_bundle`, `trunk_end_bundle`, `trunk_bundle_in_range` and so on. All these types of core bundle-mgmt APIs are quite amenable to CTests, where build small isolated test cases to just exercise and validate that these functions return the 'right thing' for a given set of inputs. 

That model of CTest-dev ... we may have to taken on in future dev-items, so that we start building a corpus of test-code to validate these lowermost-level APIs. (We can talk off-line ... w/ @rtjohnso )

Reply:
This came up on the `data-recovery` feature branch, when I was running a relatively straightforward performance test:
```
./bin/driver_test splinter_test --perf --max-async-inflight 0 --num-insert-threads 8 --num-lookup-threads 0 --num-range-lookup-threads 0 --tree-size-gib 80 --db-location /dev/nvme0n1 --set-O_DIRECT --unset-O_CREAT --cache-capacity-gib 20 --set-hugetlb --set-mlock --db-capacity-gib 120
```

I haven't recreated the issue on `main`, but the bugs definitely exist there.

All of these functions are artifacts of the current node format, which is rigid, cumbersome and bug-prone. @rtjohnso and I have tentatively discussed a new node format which will hopefully get rid of a lot of these accounting functions. I think we should punt on these tests until then. Let's have a meeting to discuss this in the not-too-distant future.